sample_code/src/fst_reference/block/vcdata.py

```diff
@@ -103,7 +104,52 @@ def _parse_position_data(position_data: bytes) -> list[int]:
     return positions
 
 
-def _parse_wave_data(wave_data: bytes, positions: list[int]) -> list[dict[str, Any]]:
+def _decompress_wave_data(data: bytes, uncompressed_length: int, waves_packtype: int, var_idx: int) -> tuple[str, dict[str, str]]:
+    """
+    Decompress wave data based on the packtype.
+    Returns (hex_data, error_dict).
+    """
+    if waves_packtype == ord("Z") or waves_packtype == ord("!"):
+        # zlib compressed
+        try:
+            dec_data = zlib.decompress(data)
+            if len(dec_data) != uncompressed_length:
+                raise RuntimeError(
+                    f"CallVCDATA: wave data uncompressed length mismatch for var {var_idx}"
+                )
+            return dec_data, {"zlib_error": ""}
+        except Exception as e:
```

Comment:
I presume the original code (which I didn't write) recorded errors like this because it always assumed lz4, which is incorrect. Now that you've fixed this does it actually fail? If not I think we should just let the exceptions propagate. That also has the side benefit of reducing the use of dictionaries.

Reply:
Let me get back you after the new year.